feat: 광고 제보 모달 UI 구현

faber-h opened the pull request:
### ✨ 이슈 번호

- #5 

### 📌 설명

- 광고 제보 모달 UI를 구현합니다.

### 📃 작업 사항

- [x] 광고 제보 모달 UI 
  - [x] 모달 컴포넌트
    - [x] 경고 svg 파일, title, subtitle
  - [x] 체크박스 컴포넌트
          1. 광고입니다. 2. 광고가 아닌데 채널이 이동했습니다.
  - [x] 체크박스 선택했을 때 하위 체크 박스 토글
          1 클릭 -> 다른 채널로 이동하기, 현재 채널 계속 시청하기
          2클릭 -> 이전 채널로 이동하기, 현재 채널 계속 시청하기
  - [x] 버튼 컴포넌트
    - [x] 취소, 확인

### 💡 참고 사항

아래는 체크박스 선택했을 때 영상입니다.

https://github.com/user-attachments/assets/e570e0d9-441e-49ca-a6b8-40ca4c3111e7

### 💭 리뷰 사항 반영

- [x] 불필요한 `<div>` 제거
- [x] 불필요한 래핑 제거
- [x] CLS 개선

### ✅ Pull Request 체크 사항

- [x] 가장 최신 브랜치를 pull 하였습니다.
- [x] base 브랜치명을 확인하였습니다.
- [x] 코드 컨벤션을 모두 확인하였습니다.
- [x] 브랜치명을 확인하였습니다.

faber-h committed:
feat: 광고 제보 모달 UI 및 로직 구성

faber-h committed:
refactor: 버튼 컴포넌트 분리

faber-h committed:
style: css 스타일 변경

faber-h committed:
refactor: map 렌더링으로 변경
- AD_REPORT_OPTIONS에 고유 id 추가
- map을 활용한 렌더링으로 코드 간결화

devjaeyoon commented on src/assets/svgs/icon-warning-red.svg:
svg의 파일명을 `icon-warning-red`라고 한 이유가 있나요? 전 Pascal 케이스로 작성했는데 통일하던가 케밥케이스가 맞다면 케밥케이스로 수정하는 것도 좋을 것 같아요!

faber-h replied:
svg나 이미지 파일에는 케밥케이스로 적는 경우가 많아 케밥 케이스로 작성했습니다.
[tailwindlabs](https://github.com/tailwindlabs/heroicons/tree/master/src/16/solid)
하지만 react 컴포넌트로 변환되는 파일이기 때문에 파스칼 케이스로 작성하는 것도 괜찮은 것 같습니다~

@Eun0713 다은님의 의견은 어떠신가요?

d0-neii replied:
저도 찾아보니까 SVG나 이미지 파일은 케밥 케이스로 작성하는 게 일반적인 것 같습니다!
희주님이 작성하신 것처럼 SVG 파일 명은 케밥 케이스로 두고 `import WarningIcon from "@/assets/svgs/icon-warning-red.svg?react"`처럼  import해서 사용할 때는 파스칼 케이스로 이름을 바꿔서 사용하는 걸로 통일하면 어떨까요 :)

faber-h replied:
저는 좋습니다!

devjaeyoon replied:
저도 좋아요! 다음 작업날에 리팩토링 해요!

faber-h replied:
[전체 스타일 정리](https://github.com/Radio-Premium/RadioPremium-FE/issues/29) 해당 issue에서 SVG 파일 명이 케밥 케이스로 통일되었습니다!

faber-h committed:
chore: 테스트용으로 수정된 Home 컴포넌트 원상 복구

devjaeyoon commented on src/components/AdReportModal.jsx:
```diff
@@ -0,0 +1,66 @@
+import Button from "./ui/Button";
+import Modal from "./ui/modal";
+
+const AD_REPORT_TYPES = {
```
상수로 관리해주시는 값이면 별도의 `constants.js` 파일을 만들어 관리해주는 것도 좋을 것 같아요!

faber-h replied:
좋은 의견 감사합니다!
현재 이 상수는 AdReportModal.jsx 내부에서만 사용되고 있어서 일단은 파일 내부에 두었습니다.
만약 추후에 다른 컴포넌트에서도 이 데이터를 공유하거나 재사용할 일이 생기면, 그때 분리해서 관리하는 것도 고려해보겠습니다~

d0-neii commented on src/components/AdReportModal.jsx:
```diff
@@ -0,0 +1,66 @@
+        )}
+      </div>
+      <div className="flex justify-end gap-x-2">
+        <Button className="rounded-md border border-gray-300 bg-white px-4 py-2 text-sm text-black hover:bg-gray-100">
```
버튼에 핸들러가 연결되어 있지 않은데 그 이유가 궁금합니다~

faber-h replied:
좋은 질문 감사합니다~
현재 버튼은 정적 구현만 완료하고 실제 동작은 아직 구현 전이라 핸들러를 연결하지 않았습니다.
추후 기능 구현 시 추가될 예정입니다~

faber-h replied:
[광고 제보 등록 API 연동](https://github.com/Radio-Premium/RadioPremium-FE/issues/60) 해당 issue에서 버튼 클릭 이벤트 핸들러가 연결되었습니다!

d0-neii commented on src/components/ToggleCheckbox.jsx:
```diff
@@ -0,0 +1,27 @@
+        {parentOption}
+      </Checkbox>
+      {isSelected && (
+        <div>
```
바깥쪽 `<div>`는 구조적으로 불필요해 보이는 거 같습니다!

faber-h replied:
네, 말씀해주신 부분 맞습니다!
해당 `<div>`는 구조상 불필요해서 제거해도 무방할 것 같아요.
리팩토링해보겠습니다. 피드백 감사합니다~

devjaeyoon commented on src/components/ui/Checkbox.jsx:
```diff
@@ -0,0 +1,10 @@
+const Checkbox = ({ children, checked, onChange }) => {
+  return (
+    <label className="flex cursor-pointer gap-2">
+      <input type="checkbox" checked={checked} onChange={() => onChange()} />
```
`onChange={() => onChange()}`를 `onChange={onChange}`로 변경해도 되지 않을까요? 사용하신 이유가 궁금합니다.

faber-h replied:
`onChange={() => onChange()}`는 불필요한 래핑이라 `onChange={onChange}`로 바로 전달해도 동일하게 작동하고 더 깔끔할 것 같습니다.
특별한 이유가 있었던 건 아니고 초기 작성할 때 습관적으로 작성한 부분인데, 말씀해주신 대로 수정하는 게 좋을 것 같아요.
감사합니다!

d0-neii commented on src/components/ui/Modal.jsx:
```diff
@@ -0,0 +1,20 @@
+        <div className="pb-2">
+          <WarningIcon className="mx-auto my-0" />
+        </div>
+        <div>
```
현재 `<div>`에 스타일 같은 역할이 없어 `<></>` 로 대체해도 괜찮을 거 같습니다!

faber-h replied:
맞아요! 해당 `<div>`는 구조적으로 필요 없는 요소라 `<></>`로 대체해도 무방할 것 같네요.
피드백 감사합니다!

faber-h committed:
refactor: 불필요한 래핑, div 제거

faber-h committed:
style: css 스타일 변경

faber-h committed:
feat: 광고로 인한 채널이 변경 여부에 따라 선택 목록 변경

faber-h committed:
feat: ToggleCheckbox에서 상위 체크 해제 시 하위 선택 초기화 기능 추가

faber-h committed:
style: 모달 그림자 추가

devjaeyoon reviewed (changes requested):
![IMG_8429](https://github.com/user-attachments/assets/c88a3478-0d5a-4feb-b417-d163fc8457ad)

모바일 환경에서 `width`가 데스크탑 환경보다 좁아 요소들이 예쁘게 보이지 않고 있어요! `min-width` 설정을 통해 모바일 환경에서도 요소들이 예쁘게 보일 수 있도록 보완 부탁드립니다.

또한, 현재 광고입니다와 광고가 아닌데 채널이 이동했습니다를 누를 경우 모달의 높이가 커지고 취소, 확인 버튼들의 요소가 아래로 밀리는 현상이 발생하고 있습니다. 이를 `CLS`라고 부르는데, `CLS`는 웹 사이트 성능을 측정할 때 감점되는 요인이라 최소 높이를 설정하고 취소, 확인 버튼들의 `position`을 아래에 고정시킴으로써 보완하면 좋을 것 같습니다.

ref. [CLS](https://web.dev/articles/cls?hl=ko)

faber-h committed:
style: CLS 개선 및 반응형 대응을 위한 modal 크기 제어
- 옵션 개수에 따라 modal min-height를 동적으로 설정하여 CLS(Cumulative Layout Shift) 현상 완화
- 모바일 환경에서도 최소 너비 확보 및 반응형 대응
- 모달 title과 checkbox 시작점 동일하도록 변경

devjaeyoon commented on src/components/AdReportModal.jsx:
```diff
@@ -0,0 +1,71 @@
+  },
+};
+
+const AdReportModal = ({ isChangeChannel }) => {
```
`광고가 나오는데 채널이 이동하지 않은 경우`와 `광고가 나오지 않았는데 채널이 이동한 경우`를 구분하기 위한 변수명은 채널 이동 여부 변수명은 `isChangeChannel`보다 `isChanneldChanged` 어떨까요?

devjaeyoon commented on src/components/AdReportModal.jsx:
```diff
@@ -0,0 +1,71 @@
+  };
+
+  const optionsToRender = Object.entries(AD_REPORT_OPTIONS).filter(([key]) => {
+    if (!isChangeChannel) return true;
```
```suggestion
    if (!isChangeChannel) {
      return true;
    }
```

devjaeyoon commented on src/components/AdReportModal.jsx:
```diff
@@ -0,0 +1,71 @@
+
+  const optionsToRender = Object.entries(AD_REPORT_OPTIONS).filter(([key]) => {
+    if (!isChangeChannel) return true;
+    return key === AD_REPORT_TYPES.AD;
```
return 사이에 개행 넣어주세요!

devjaeyoon reviewed (changes requested):
리뷰 확인 부탁드립니다!

faber-h committed:
refactor: 변수, 함수명 변경

faber-h committed:
refactor: import문 절대경로로 수정

Eun0713 reviewed (approved):
정말 수고하셨습니다! LGTM✨

devjaeyoon reviewed (approved):
고생하셨습니다! LGTM!
